agent/config/parse_windows.go

```diff
@@ -113,3 +123,61 @@ var isWindows2016 = func() (bool, error) {
 
 	return false, nil
 }
+
+func getInstallationType(installationType string) (string, error) {
+	if strings.Compare(installationType, InstallationTypeFull) == 0 {
+		return "FULL", nil
+	} else if strings.Compare(installationType, InstallationTypeCore) == 0 {
+		return "CORE", nil
+	} else {
+		err := seelog.Errorf("Invalid Installation type")
+		return "", err
+	}
+}
+
+func GetOSFamilyType() string {
+	osTypeFormat := "WINDOWS_SERVER_%s_%s"
+	key, err := registry.OpenKey(registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\Windows NT\CurrentVersion`, registry.QUERY_VALUE)
+	if err != nil {
+		seelog.Errorf("Unable to open Windows registry key to determine Windows version: %v", err)
+		return UnknownOSType
+	}
+	defer key.Close()
+
+	productName, _, err := key.GetStringValue("ProductName")
+	if err != nil {
+		seelog.Errorf("Unable to read registry key, ProductName: %v", err)
+		return UnknownOSType
+	}
+	installationType, _, err := key.GetStringValue("InstallationType")
+	if err != nil {
+		seelog.Errorf("Unable to read registry key, InstallationType: %v", err)
+		return UnknownOSType
+	}
+	iType, err := getInstallationType(installationType)
+	if err != nil {
+		seelog.Errorf("Invalid Installation type found: %v", err)
+		return UnknownOSType
+	}
+	releaseId, _, err := key.GetStringValue("ReleaseId")
+	if err != nil {
+		seelog.Errorf("Unable to read registry key, ReleaseId: %v", err)
+		return UnknownOSType
+	}
+
+	if strings.HasPrefix(productName, Windows_Server_2019) {
+		osTypeFormat = fmt.Sprintf(osTypeFormat, "2019", iType)
```

Comment:
I find it confusing that you are defining this string as a format string (with `%s`), and then overwriting it as a regular string that ultimately gets returned.

You should create a new variable for the first part of this string similar to how you do for `iType`. In these if-statements, only set the variable to that number. As in: 
```
num := "2019"
```

then you can have just a single return like `return fmt.Sprintf("WINDOWS_SERVER_%s_%s", num, iType)`